features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/resources/ResourceDataStoreManager.kt

```diff
@@ -0,0 +1,159 @@
+
+                val lastUpdateDateNs = storedData.lastUpdateDateNs
+                val storedHashes = storedData.resourceHashes
+
+                storedLastUpdateDateNs.set(lastUpdateDateNs.toLong())
+
+                if (didDataStoreExpire()) {
+                    handleDataStoreExpired(
+                        storedHashes = storedHashes.toSet(),
+                        callback = onDeleteDatastoreCallback
+                    )
+                } else {
+                    knownResources.addAll(storedHashes)
+                    finishedInitializingManager()
+                }
+            },
+            onFetchFailure = {
+                // No need to do anything here.
+                // The next time we write to the datastore it will overwrite
+            }
+        )
+    }
+
+    internal fun isPreviouslySentResource(resourceHash: String): Boolean =
+        knownResources.contains(resourceHash)
+
+    internal fun cacheResourceHash(resourceHash: String) {
+        knownResources.add(resourceHash)
+        writeResourcesToStore()
+    }
+
+    internal fun isReady(): Boolean =
+        isInitialized.get()
+
+    // region internal
+
+    private fun finishedInitializingManager() {
+        isInitialized.set(true)
+    }
+
+    private fun writeResourcesToStore() {
+        val data = ResourceHashesEntry(
+            lastUpdateDateNs = storedLastUpdateDateNs,
+            resourceHashes = knownResources.toList()
+        )
+
+        featureSdkCore.getFeature(
+            SESSION_REPLAY_RESOURCES_FEATURE_NAME
+        )?.dataStore?.setValue(
+            data = data,
+            key = DATASTORE_HASHES_ENTRY_NAME,
+            callback = object : DataStoreWriteCallback {
+                override fun onSuccess() {
+                    // no need to do anything here
+                }
+
+                override fun onFailure() {
+                    // no need to do anything here
+                }
+            },
+            serializer = resourceHashesSerializer
+        )
+    }
+
+    private fun fetchStoredResourceHashes(
+        onFetchSuccessful: (dataStoreContent: DataStoreContent<ResourceHashesEntry>?) -> Unit,
+        onFetchFailure: () -> Unit
+    ) {
+        featureSdkCore.getFeature(
+            SESSION_REPLAY_RESOURCES_FEATURE_NAME
+        )?.dataStore?.value(
+            key = DATASTORE_HASHES_ENTRY_NAME,
+            deserializer = resourceHashesDeserializer,
+            callback = object : DataStoreReadCallback<ResourceHashesEntry> {
+                override fun onSuccess(dataStoreContent: DataStoreContent<ResourceHashesEntry>?) {
+                    onFetchSuccessful(dataStoreContent)
+                }
+
+                override fun onFailure() {
+                    onFetchFailure()
+                }
+            }
+        )
+    }
+
+    private fun handleDataStoreExpired(storedHashes: Set<String>, callback: DataStoreWriteCallback) {
+        knownResources.removeAll(storedHashes)
```

Comment:
do you need this line? `handleDataStoreExpired` is called always when `isReady` is `false`, meaning `knownResources` is always empty at this point.